pygmt/helpers/decorators.py

```diff
@@ -338,9 +357,15 @@ def new_module(*args, **kwargs):
                     value = kwargs[arg]
                     issequence = fmt in separators
                     if issequence and is_nonstr_iter(value):
-                        kwargs[arg] = separators[fmt].join(
-                            "{}".format(item) for item in value
-                        )
+                        for index, item in enumerate(value):
+                            try:
+                                assert " " not in str(item)
```

Comment:
I'm a little confused how the `assert` statement works. Could you please explain a little bit?

Reply:
Sorry that this isn't very readable, it's following the [EAFP style](https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python#11360880), usually used when we're handling only a few exceptional cases. The original reason that we can't pass in 'datetime'-like values to the 'region' argument is because there's a space `" "` when converting a `pandas.Timestamp` to a string (instead of a `"T"` that would make it ISO compliant).

```python
import pandas as pd

str(pd.Timestamp("2015-01-01T12:00:00.123456789"))
# '2015-01-01 12:00:00.123456789'
```

This also applies to `xarray.DataArray` datetime values. So the `assert` here checks for the space `" "`, and if it finds it, it will try to convert the item into an ISO compliant datetime value (e.g. `"2015-01-01T12:00:00.123456789"`).